pkg/machine/hyperv/stubber.go

```diff
@@ -497,12 +496,15 @@ func readAndSplitIgnition(mc *vmconfigs.MachineConfig, vm *hypervctl.VirtualMach
 }
 
 func removeIgnitionFromRegistry(vm *hypervctl.VirtualMachine) error {
-	pairs, err := vm.GetKeyValuePairs()
-	if err != nil {
-		return err
-	}
-	for key := range pairs {
-		if err := vm.RemoveKeyValuePair(key); err != nil {
+	// because the vm is down at this point, we cannot query hyperv for these key value pairs.
+	// therefore we blindly iterate from 0-50 and delete the key/value pairs. hyperv does not
+	// raise an error if the key is not present
+	//
+	for i := 0; i < 50; i++ {
+		// this is a well known "key" defined in libhvee and is the vm name
+		// plus an index starting at 0
+		key := fmt.Sprintf("%s%d", vm.ElementName, i)
+		if err := vm.RemoveKeyValuePairNoWait(key); err != nil {
```

Comment:
Ok not like I undersatnd the windows code so consider this non blocking.

First where does this 50 came from? How do we know this is enough or wont change?

Second why is it our task to remove hyper registry keys here, I would think this should be done by the vm.Remove() call in libhvee?

Reply:
the 50 is related to the number of 1023 byte chunks needed to describe the ignition file.  windows registry values can only be 1023 bytes long.  So, we had to pick a number because you cannot query the vm for how many keys it has when the vm is down ... that is why we had the error before

as for libhvee, it was meant to be a generic library, not something strictly for podman ... i would not object to long term having this code live in there and it just be an API call.  But for now, and 5.0.1 plans, this fixes things and puts it behind us.